crates/tx-parser/src/filter.rs

```diff
@@ -24,6 +25,77 @@ pub enum TxFilterRule {
     DepositRequest(DepositTxParams),
     /// Deposit transaction with deposit config and address
     Deposit(DepositTxParams),
+    /// Addresses that are spent to
+    SpentToAddrs(Vec<BitcoinAddress>),
+    /// Blob ids that are expected
+    BlobIds(Vec<Buf32>),
+    /// Outpoints
+    Outpoints(Vec<(Buf32, u32)>),
+}
+
+#[derive(Clone, Debug)]
+pub struct TxFilterConfig {
+    /// For checkpoint update inscriptions.
+    rollup_name: RollupName,
+
+    /// For addresses that we expect spends to.
+    expected_addrs: Vec<BitcoinAddress>,
+
+    /// For blobs we expect to be written.
+    expected_blobs: Vec<Buf32>,
+
+    /// For deposits that might be spent from.
+    expected_outpoints: Vec<(Buf32, u32)>,
+    // eventually, in future version
+    // For bridge V1 deposits that do bitmap flagging for the multisig addr.
+    // operator_key_tbl: PublickeyTable,
+    /// EE addr length
+    ee_addr_len: u8,
+
+    /// Deposit denomination
+    deposit_denomination: u64, // sats
+
+    /// Operators addr
+    operator_addr: BitcoinAddress,
```

Comment:
Possibly. What addresses do you suppose client will be watching for besides the operators multisig address? Parsing txs that spend to different address might result in different ProtocolOps. So, I am thinking of doing something like:
```rust
pub struct ExpectedAddr {
   address: Address,
   // type will indicate what exactly we need to parse from the transaction so that we can have all addrs sorted in a vec
   type: Operator | ...
}

// This one for sorting the addrs
impl PartialOrd for ExpectedAddr {
   fn partial_cmp() { // compare address }
}
```